save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/vulnerability/VulnerabilityController.kt

```diff
@@ -247,15 +245,38 @@ class VulnerabilityController(
         summary = "Save new date.",
         description = "Save new date.",
     )
+    @PreAuthorize("permitAll()")
     @ApiResponse(responseCode = "200", description = "Successfully saved new date in vulnerability")
+    @ApiResponse(responseCode = "404", description = "Could not either find user or find vulnerability")
+    @ApiResponse(responseCode = "409", description = "Could not save date as it is already present in vulnerability")
     fun saveDate(
         @RequestBody vulnerabilityDateDto: VulnerabilityDateDto,
         authentication: Authentication,
     ): Mono<StringResponse> = blockingToMono {
         vulnerabilityService.saveDate(vulnerabilityDateDto, authentication)
-    }.map {
-        ResponseEntity.ok("Date was successfully saved in vulnerability")
     }
+        .switchIfErrorToConflict {
+            "Date with type ${vulnerabilityDateDto.type} is already present in ${vulnerabilityDateDto.vulnerabilityName} vulnerability"
+        }
+        .map { ResponseEntity.ok("Date was successfully saved in vulnerability") }
+
+    @PostMapping("/delete-date")
+    @Operation(
+        method = "POST",
+        summary = "Delete date.",
+        description = "Delete date.",
+    )
+    @PreAuthorize("permitAll()")
+    @ApiResponse(responseCode = "200", description = "Successfully deleted date in vulnerability")
+    @ApiResponse(responseCode = "404", description = "Requested date is not found")
+    @Suppress("UnusedParameter")
+    fun deleteDate(
+        @RequestBody vulnerabilityDateDto: VulnerabilityDateDto,
```

Comment:
I guess do you use `post` because of `request body`?

Reply:
Yes